src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -118,5 +122,10 @@ private bool ValidateSignature(Signature? signature, IReleaseSpec spec)
 
             return !spec.ValidateChainId || (signature.V == 27 || signature.V == 28);
         }
+
+        private bool Validate4844Fields(Transaction transaction, IReleaseSpec spec)
+        {
+            return transaction.Type != TxType.Blob || spec.IsEip4844Enabled ^ transaction.MaxFeePerDataGas is null;
```

Comment:
hmmm are you sure that this condition is correct? adding TxValidatorTests might help you